src/filters/fb_network.rs

```diff
@@ -0,0 +1,307 @@
+            o.sort_unstable();
+            o.dedup();
+            self.builder.create_vector(&o)
+        });
+
+        let opt_not_domains = network_filter.opt_not_domains.as_ref().map(|v| {
+            let mut o: Vec<u16> = v.iter().map(|x| self.get_or_insert(x)).collect();
+            o.sort_unstable();
+            o.dedup();
+            self.builder.create_vector(&o)
+        });
+
+        let modifier_option = network_filter
+            .modifier_option
+            .as_ref()
+            .map(|s| self.builder.create_string(&s));
+
+        let hostname = network_filter
+            .hostname
+            .as_ref()
+            .map(|s| self.builder.create_string(&s));
+
+        let tag = network_filter
+            .tag
+            .as_ref()
+            .map(|s| self.builder.create_string(&s));
+
+        let patterns = if network_filter.filter.iter().len() > 0 {
+            let offsets: Vec<WIPOffset<&str>> = network_filter
+                .filter
+                .iter()
+                .map(|s| self.builder.create_string(s))
+                .collect();
+            Some(self.builder.create_vector(&offsets))
+        } else {
+            None
+        };
+
+        let raw_line = network_filter
+            .raw_line
+            .as_ref()
+            .map(|v| self.builder.create_string(v.as_str()));
+
+        let filter = fb::NetworkFilter::create(
+            &mut self.builder,
+            &fb::NetworkFilterArgs {
+                mask: network_filter.mask.bits(),
+                patterns: patterns,
+                modifier_option: modifier_option,
+                opt_domains: opt_domains,
+                opt_not_domains: opt_not_domains,
+                hostname: hostname,
+                tag: tag,
+                raw_line: raw_line,
+            },
+        );
+
+        self.filters.push(filter);
+        u32::try_from(self.filters.len() - 1).expect("< u32::MAX")
+    }
+
+    pub fn finish(&mut self) -> Vec<u8> {
+        let filters = self.builder.create_vector(&self.filters);
+
+        let unique_domains = self.builder.create_vector(&self.unique_domains);
+
+        let storage = fb::NetworkFilterList::create(
+            &mut self.builder,
+            &&fb::NetworkFilterListArgs {
+                network_filters: Some(filters),
+                unique_domains_hashes: Some(unique_domains),
+            },
+        );
+        self.builder.finish(storage, None);
+
+        let binary = Vec::from(self.builder.finished_data());
+        binary
+    }
+}
+pub struct FlatPatterns<'a> {
+    patterns: Option<flatbuffers::Vector<'a, flatbuffers::ForwardsUOffset<&'a str>>>,
+}
+
+impl<'a> FlatPatterns<'a> {
+    #[inline(always)]
+    pub fn new(
+        patterns: Option<flatbuffers::Vector<'a, flatbuffers::ForwardsUOffset<&'a str>>>,
+    ) -> Self {
+        Self { patterns }
+    }
+
+    #[inline(always)]
+    pub fn iter(&self) -> FlatPatternsIterator {
```

Comment:
It's not absolutely clear why we implement iterator traits instead using the structure as-is.
If we do it to use a one `check_pattern()` for both "old" and flat impl, let's make a comment to inline this after "old" impl is gone.

Reply:
because I removed direct dependency from any storages in the check functions. I'm not planning to inline it